File: training-data/ruby/insertion_sort.rb
def insertion_sort(items)
  arr = items.dup
  (1...arr.length).each do |i|
    key = arr[i]
    j = i - 1
    while j >= 0 && arr[j] > key
      arr[j + 1] = arr[j]
      j -= 1
    end
    arr[j + 1] = key
  end
  arr
end

puts insertion_sort([5, 3, 8, 1, 9, 2]).inspect

File: training-data/dart/insertion_sort.dart
List<int> insertionSort(List<int> arr) {
  final a = List<int>.from(arr);
  for (var i = 1; i < a.length; i++) {
    final key = a[i];
    var j = i - 1;
    while (j >= 0 && a[j] > key) {
      a[j + 1] = a[j];
      j--;
    }
    a[j + 1] = key;
  }
  return a;
}

void main() {
  print(insertionSort([12, 11, 13, 5, 6]));
}
